lib/utils/RequestHandler.js

```diff
@@ -238,6 +238,14 @@ class RequestHandler {
             })
         })
     }
+
+    extendDefaultOptions (opts) {
+        const shouldExtend = Object.keys(opts).some((key) => opts[key] !== this.defaultOptions[key])
```

Comment:
why do we need to check this stuff? Why not just update `this.defatultOptions` unconditionally?